plugin/helpers_bots.go

```diff
@@ -4,12 +4,84 @@
 package plugin
 
 import (
+	"io/ioutil"
+	"path/filepath"
+
 	"github.com/mattermost/mattermost-server/model"
 	"github.com/mattermost/mattermost-server/utils"
 	"github.com/pkg/errors"
 )
 
-func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error) {
+type ensureBotOptions struct {
+	ProfileImagePath string
+	IconImagePath    string
+}
+
+type EnsureBotOption func(*ensureBotOptions)
+
+func ProfileImagePath(path string) EnsureBotOption {
+	return func(args *ensureBotOptions) {
+		args.ProfileImagePath = path
+	}
+}
+
+func IconImagePath(path string) EnsureBotOption {
+	return func(args *ensureBotOptions) {
+		args.IconImagePath = path
+	}
+}
+
+func (p *HelpersImpl) readImage(path string) ([]byte, error) {
+	bundlePath, err := p.API.GetBundlePath()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get bundle path")
+	}
+
+	imageBytes, err := ioutil.ReadFile(filepath.Join(bundlePath, path))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to read image")
+	}
+	return imageBytes, nil
+}
+
+func (p *HelpersImpl) setIconAndProfileImageIfNotEmpty(botId string, profileImagePath string, iconImagePath string) error {
+	if !(profileImagePath == "") {
+		imageBytes, err := p.readImage(profileImagePath)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read profile image")
+		}
+		setProfileErr := p.API.SetProfileImage(botId, imageBytes)
+		if setProfileErr != nil {
+			return errors.Wrap(err, "Failed to set profile image")
+		}
+	}
+	if !(iconImagePath == "") {
+		imageBytes, err := p.readImage(iconImagePath)
+		if err != nil {
+			return errors.Wrap(err, "Failed to read icon image")
+		}
+		setIconErr := p.API.SetBotIconImage(botId, imageBytes)
+		if setIconErr != nil {
+			return errors.Wrap(err, "Failed to set profile image")
+		}
+	}
+	return nil
+}
+
+// EnsureBot either returns an existing bot user matching the given bot, or creates a bot user from the given bot.
+// Returns the id of the resulting bot. A profile image or icon image may be optionally passed in to be set for
+// the existing or newly created bot.
```

Comment:
I've might have been ambiguous here: The comment should be in `plugin/helpers.go`. You don't have to add a comment here.